### DISCUSSION TITLE
Fluidpatcher on Mac

### DISCUSSION THREAD

**Discussion by Payou67** [GeekFunkLabs/fluidpatcher#92] [Q&A] [answered]
Hi!
I'm trying to start the desktop version of Fluidpatcher on a Mac but in Terminal I get this error
`- SyntaxError: invalid syntax (__init__.py, line 173)`
Any clues?

**Comment by albedozero** [comment]
If you're using the latest version that should be [this line](https://github.com/GeekFunkLabs/fluidpatcher/blob/8c953ad3a56424f2dcb8ab937b945ee117d6bf47/fluidpatcher/__init__.py#L173C71-L173C71), and I don't see any issues. Was the entire text of the error? What python version are you using (`python --version`)?

**Reply by Payou67** [reply]
`Python 3.8.3 (v3.8.3:6f8c8320e9, May 13 2020, 16:29:34) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
`
I'm a Newbie at Python so I've tried in Terminal this command: python3 fluidpatcher.pyw
and this is what I get:
`File "/Users/pascalsauvageau/fluidpatcher/__init__.py", line 173
    for zone in self.bank, *self.bank.get('patches', {}).values():`

**Reply by albedozero** [reply, marked as answer]
Ah - it looks like Python 3.8 needs parentheses around tuples in `for` statements. I haven't made it official, but I made some coding choices a bit ago that made this project incompatible with Python <3.9.  Your version is getting a bit old (May 2020) so you may want to upgrade anyway.

**Comment by Payou67** [comment]
Allright I’ll update my Python

Thanks Bill

Envoyé de mon iPhone
